### PR TITLE
Node14 Serverless 3 upgrade

### PR DESCRIPTION
**Pull request by nnti**
## Purpose

Upgraded 508 to serverless version 3 and upgraded the runtime node 14.

#### Linked Issues to Close

https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/216?modal=detail&selectedIssue=OY2-17896
https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/216?modal=detail&selectedIssue=OY2-18071

## Approach

- Serverless framework was updated from version 2 to version 3 in the top level package.json.
- Runtime was updated from version 12 to version 14
- As many plugins in the sls plugin ecosystem made updates to accommodate v3, all sls plugins were updated to latest ( yarn upgrade pluginname --latest ). No issues were found.
- Deployment and app functionality was verified to work as intended.

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Comment by kenrickveit** [comment]
@nnti now that we are moving this to serverless v3, can you please try removing this line where I was specifying the v3 variable resolver?
[variablesResolutionMode: "20210326"](https://github.com/CMSgov/macpro-platform-doc-conversion/blob/7e9eae4aa003daec5b38eee003e58661fd63c910/services/app-api/serverless.yml#L5)

File: package.json [hunk 1]
```diff
@@ -17,15 +17,15 @@
...
     "serverless-online": "CMSgov/serverless-online",
     "serverless-plugin-scripts": "^1.0.2",
-    "serverless-plugin-warmup": "^5.3.1",
+    "serverless-plugin-warmup": "^7.1.0",
     "serverless-s3-bucket-helper": "CMSgov/serverless-s3-bucket-helper#0.1.1",
-    "serverless-stack-termination-protection": "^1.0.4"
+    "serverless-stack-termination-protection": "^2.0.0"
   },
   "dependencies": {},
   "//resolutions": {
```
Comment (kenrickveit) [inline package.json:31]: I had to pin these dependencies to close vulnerabilities as the versions that were required with Serverless v2 were too low.  Can you please remove this resolutions block and see if it still builds ok?
We can see if any security issues re-open post merge.

Comment (kenrickveit) [reply package.json:31]: IE lines 31-38 can probably come out entirely now.

**Comment by kenrickveit** [comment]
@nnti the reason deploy-support failed after you merged this before is that we don't have a few variables set in SSM, and now in v3 it fails instead of warning on this issue.

@mdial89f do we use a Github PAT for this?
https://github.com/CMSgov/macpro-platform-doc-conversion/blob/master/services/.sechub/serverless.yml#L51

**Comment by mdial89f** [comment]
> @nnti the reason deploy-support failed after you merged this before is that we don't have a few variables set in SSM, and now in v3 it fails instead of warning on this issue.
> 
> @mdial89f do we use a Github PAT for this? https://github.com/CMSgov/macpro-platform-doc-conversion/blob/master/services/.sechub/serverless.yml#L51

Yep

**Comment by kenrickveit** [comment]
> > @nnti the reason deploy-support failed after you merged this before is that we don't have a few variables set in SSM, and now in v3 it fails instead of warning on this issue.
> > Mike Dial do we use a Github PAT for this? https://github.com/CMSgov/macpro-platform-doc-conversion/blob/master/services/.sechub/serverless.yml#L51
> 
> Yep

Thank you sir.  @nnti can you try generating a PAT and adding these vars in SSM in the platform dev account please?

**Review by kenrickveit — CHANGES_REQUESTED** [review]
@nnti I've finished the review. Please make the requested changes and re-test they example py script is giving us a good pdf.

Thanks!

**Comment by nnti** [comment]
ok I've added the values for githubRepository, githubRepositoryProjects, and githubOrganizationProjects in ssm. Let me know the expiration timeframe and scopes for the PAT. 
Also, I'm not sure what value I need for severity.

**Comment by nnti** [comment]
@kenrickveit  Requested updates have been made. Let me know what to do for the githubAccessToken

**Comment by kenrickveit** [comment]
> @kenrickveit Requested updates have been made. Let me know what to do for the githubAccessToken

Nathan,
On the token expiration, this will be your token, so it is up to you. I'd prefer no expiration or at least one year, as we'll need to rotate it. 

On scope, I'm not certain, I'd try `repo` and we could adjust it re-run deploy-support after.

@mdial89f since you originally created this in the quickstart, do you happen to know what scope we need here for the PAT to be sufficient for the `.sechub` service?
https://github.com/CMSgov/macpro-quickstart-serverless/pull/319

@nnti please confirm you have re-tested the Python generation of the pdf since the last changes, and I'll approve.  We can always re-run deploy-support with the correct params in SSM after the fact.

**Comment by nnti** [comment]
> > @kenrickveit Requested updates have been made. Let me know what to do for the githubAccessToken
> 
> Nathan, On the token expiration, this will be your token, so it is up to you. I'd prefer no expiration or at least one year, as we'll need to rotate it.
> 
> On scope, I'm not certain, I'd try `repo` and we could adjust it re-run deploy-support after.
> 
> @mdial89f since you originally created this in the quickstart, do you happen to know what scope we need here for the PAT to be sufficient for the `.sechub` service? [CMSgov/macpro-quickstart-serverless#319](https://github.com/CMSgov/macpro-quickstart-serverless/pull/319)
> 
> @nnti please confirm you have re-tested the Python generation of the pdf since the last changes, and I'll approve. We can always re-run deploy-support with the correct params in SSM after the fact.

@kenrickveit Done. I've generated the githubAccessToken and stored its value in ssm. Also, the python gen command successfully generated the pdf.

**Review by kenrickveit — APPROVED** [review]
Thanks @nnti, approved!

**Comment by mdial89f** [comment]
I don't remember the required scope, no.

**Pull request merged**